### ISSUE TITLE
[Feature] __builtin_setjmp and __builtin_longjmp support

### ISSUE THREAD

**Issue by raedwulf** [sustrik/libdill#69]
I believe it is possible to have support for `__builtin_setjmp` and `__builtin_longjmp` which are compiler-specific optimised implementations of the `setjmp`/`longjmp` which do not depend on glibc.  However, the downside of those are that you *should* use the same compiler with the the application and the library (but GCC and clang are compatible with each other in this instance).

It's just an idea - from my previous tests, it's slower than assembly, but faster than glibc-based ones.

It does not support having setjmp and longjmp in the same function so the two need to be in separate ones and the second one marked with `__attribute__((noinline))`.

This is a candidate for another flag and it might fill the gap between -DDILL_ARCH_FALLBACK and the assembly version.

**Comment by sustrik** [comment]
Fallback options that are almost never used are a maintenance liability. Having to support DILL_POLL and DILL_ARCH_FALLBACK is bad enough. Why add more?

**Comment by raedwulf** [comment]
That's true - I just want to keep note in case we need to use it - the patch in #18 also uses it.
